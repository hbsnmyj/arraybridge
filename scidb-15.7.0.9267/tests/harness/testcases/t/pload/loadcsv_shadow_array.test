--setup

--start-query-logging

--shell --command " cp ${TEST_DATA_DIR}/sm11_shadow.csv /tmp "

--test

--start-igdata
 
create array array_sm10 <a:int64,b:int64,c:int64>[i=0:*,4,0]
load(array_sm10, '/tmp/sm11_shadow.csv', -2, 'csv', 100, array_shadow)

remove(array_sm10)
create array array_sm10 <a:int64 NULL,b:int64,c:int64>[i=0:*,4,0]
load(array_sm10, '/tmp/sm11_shadow.csv', -2, 'csv', 100, array_shadow2)


#--shell --command " cat ${TEST_DATA_DIR}/sm11_shadow.csv | loadcsv.py -a 'array_sm10' -s '<a:int64,b:int64,c:int64>[i=0:*,4,0]' -v -x -c 4 -l -L -b -e 100 -w 'array_shadow' -d 'localhost' -p $IQUERY_PORT  "

#--shell --command " cat ${TEST_DATA_DIR}/sm11_shadow.csv | loadcsv.py -a 'array_sm10' -s '<a:int64 NULL,b:int64,c:int64>[i=0:*,4,0]' -v -x -c 4 -l -L -b -e 100 -w 'array_shadow2' -d 'localhost' -p $IQUERY_PORT "


--stop-igdata


#scan( array_sm10 )
#scan( array_shadow )
project( array_sm10, a, b, c )
project( array_shadow, a, b, c )
project( array_shadow2, a, b, c)
#scan( array_shadow2 )
--cleanup

remove(array_shadow)
remove(array_shadow2)
remove(array_sm10)
--shell --command " rm /tmp/sm11_shadow.csv "

--stop-query-logging

