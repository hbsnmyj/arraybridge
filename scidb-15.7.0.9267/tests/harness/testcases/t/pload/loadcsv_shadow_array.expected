SCIDB QUERY : <create array array_sm10 <a:int64,b:int64,c:int64>[i=0:*,4,0]>
[Query was executed successfully, ignoring data output by this query.]

SCIDB QUERY : <load(array_sm10, '/tmp/sm11_shadow.csv', -2, 'csv', 100, array_shadow)>
[Query was executed successfully, ignoring data output by this query.]

SCIDB QUERY : <remove(array_sm10)>
[Query was executed successfully, ignoring data output by this query.]

SCIDB QUERY : <create array array_sm10 <a:int64 NULL,b:int64,c:int64>[i=0:*,4,0]>
[Query was executed successfully, ignoring data output by this query.]

SCIDB QUERY : <load(array_sm10, '/tmp/sm11_shadow.csv', -2, 'csv', 100, array_shadow2)>
[Query was executed successfully, ignoring data output by this query.]

SCIDB QUERY : <project( array_sm10, a, b, c )>
{i} a,b,c
{0} null,1,1
{1} 1,2,3
{2} 4,0,6
{3} 7,8,9

SCIDB QUERY : <project( array_shadow, a, b, c )>
{i} a,b,c
{0} 'Assigning NULL to non-nullable attribute',null,null
{2} null,'Assigning NULL to non-nullable attribute',null

SCIDB QUERY : <project( array_shadow2, a, b, c)>
{i} a,b,c
{2} null,'Assigning NULL to non-nullable attribute',null

SCIDB QUERY : <remove(array_shadow)>
Query was executed successfully

SCIDB QUERY : <remove(array_shadow2)>
Query was executed successfully

SCIDB QUERY : <remove(array_sm10)>
Query was executed successfully

